server/src/main/java/org/elasticsearch/ingest/PipelineProcessor.java

```diff
@@ -19,25 +19,27 @@
 
 package org.elasticsearch.ingest;
 
+import org.elasticsearch.script.TemplateScript;
+
 import java.util.Map;
 import java.util.function.BiConsumer;
 
 public class PipelineProcessor extends AbstractProcessor {
 
     public static final String TYPE = "pipeline";
 
-    private final String pipelineName;
-
+    private final TemplateScript.Factory pipelineName;
```

Comment:
would it make sense to rename this to something that points it to the templating aspect? (eg. `pipelineTemplate` ?) it's a bit confusing otherwise below when we render the template and yield a `pipelineName` from another `pipelineName`